core/blockchain.go

```diff
@@ -559,6 +564,49 @@ func (bc *BlockChain) initHvmHeaderNode(config *tbc.Config) {
 	bc.hvmEnabled = true
 }
 
+func (bc *BlockChain) SetupDeucalion(pctx context.Context, address string) error {
+	d, err := deucalion.New(&deucalion.Config{
+		ListenAddress: address,
+	})
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithCancel(pctx)
+
+	go func() {
+		defer cancel()
+		if err := d.Run(ctx, nil, func(ctx context.Context) (bool, any, error) {
+			return bc.healthyNode.Load(), bc.CurrentBlock(), nil
+		}); !errors.Is(err, context.Canceled) {
+			log.Error("deucalion terminated with error", "err", err)
+			return
+		}
+		log.Info("deucalion clean shutdown")
+	}()
+
+	go func() {
+		defer cancel()
+		oldHeader := bc.CurrentHeader()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-time.After(progressionInterval):
+				curHeader := bc.CurrentHeader()
+				if curHeader.Number.Cmp(oldHeader.Number) <= 0 {
+					bc.healthyNode.Store(false)
+				} else {
+					bc.healthyNode.Store(true)
+				}
```

Comment:
I think this can be simplified to:
```suggestion
				// Healthy if we have progressed at least 1 block.
				bc.healthyNode.Store(curHeader.Number.Cmp(oldHeader.Number) > 0)
```

Reply:
I find the simplified version a little harder to read, but I'm fine with it either way